text/logs/0114-log-correlation.md

```diff
@@ -0,0 +1,137 @@
+result, resource correlation may not happen at the log entry level, so we may or 
+may not put the resource correlation information in the log entry itself.
+
+Resource information may be managed as part of the log ingestion process- for 
+example a Docker logging driver will know which container logs came from. 
+Full resource information may also not be available, as when logs are 
+aggregated by syslog or a similar system that has less context available to it. 
+As a result resource correlation information in the logs entries themselves 
+should be considered optional. If included, the resource information should 
+follow the semantic conventions for resources.
+
+If the log entry is expressed in key-value pairs, any resource keys should be 
+prepended with ‘resource’- for example, `resource.service.name=”shoppingcart”`. 
+If the entry is expressed in JSON, the resource key-values should be placed in 
+an object named “resource” at the top level of the object.
+
+### Correlation Context
+
+A Correlation Context is a set of key-value pairs that is shared amongst the 
+spans of a distributed trace. Like span attributes, the correlation context may 
+already be carried with span information, so duplicating this information may be 
+redundant. In certain cases it may be important to associate this context with 
+log entries. When the context is embedded in a log entry, the key-value
+pairs should be placed in a ‘correlation’ namespace. Where key-value pairs are 
+supported, embed the correlation key as “correlation.key_name”. In JSON or 
+other formats that allow nested structures, the key-value pairs should be 
+placed in an object named ‘correlation’ at the top of the object.
+
+
+## Internal details
+
+### Encoding
+#### Key-Value Pairs
+    2020-05-20 20:13:31 INFO Message logged. resource.hostname=myhost correlation.user=djones traceid=0354af75138b12921 spanid=14c902d73a traceflags=01
+#### JSON
+    {
+      "time": "2020-05-20 20:13:31",
+      "msg": "Message logged.",
+      "level": "INFO",
+      "correlation": {
+        “user”: “djones”
+      },
+      "resource": {
+        "hostname":"myhost"
+      }
+      "traceid": "0354af75138b12921",
+      "spanid": "14c902d73a",
+      "traceflags": "01"
+    }
+
+### Custom Format
+
+Custom formats that don’t allow for automatic parsing of key-value pairs can be 
+used, but they will require synchronization between the output format and the 
+extraction mechanism. These types of extractions may have the advantage of being 
+less verbose, but they also have the disadvantage of requiring setting up a 
+custom extraction process, and may be more fragile. Since this approach is 
+vendor-dependent, there is little guidance that can be provided by 
+OpenTelemetry.
+
+## Trade-offs and mitigations
+
+As a convention, adherence to these standards may have some variation, and some
+of this advice may be difficult to implement in certain circumstances.
```

Comment:
It is not quite clear what this means. What level of variation is acceptable? Typically, where it matters we use RFC kewords MAY, SHOULD, MUST, etc to specify how strict the particular part of the spec is and what variation is allowed.